api/envoy/api/v2/route/route_components.proto

```diff
@@ -137,9 +137,12 @@ message VirtualHost {
 
   // Decides whether the :ref:`x-envoy-attempt-count
   // <config_http_filters_router_x-envoy-attempt-count>` header should be included
-  // in the upstream request. Setting this option will cause it to override any existing header
+  // in the upstream request and the downstream response.
+  // Setting this option will cause it to override any existing header
   // value, so in the case of two Envoys on the request path with this option enabled, the upstream
-  // will see the attempt count as perceived by the second Envoy. Defaults to false.
+  // will see the attempt count as perceived by the second Envoy. Similarly, if there are multiple Envoys
+  // on the response path with this option enabled, the downstream will see the attempt count
+  // as perceived by the first Envoy in the path. This field Defaults to false.
```

Comment:
This seems a data path visible behavioral change for users of the existing API. Should there be a new field for downstream header addition?

Reply:
Yeah, I think we either runtime guard this or have a separate knob

Reply:
👌 that is what I suspected per 1. above. What I am more indecisive about is having the separate knob vs using the same knob with a runtime guard that goes through the release cycle. Any preference from the two of you @alyssawilk @htuch?

Reply:
Separate knob IMHO. Runtime guarding makes the API behavior unstable within a major version, which we don't allow.

Reply:
Just for clarity here, if we thought it belonged as one knob long term, the correct thing to do would be to have a new name for "upstream and downstream knob" and deprecate the old one, right?
I'm not hearing any strong opinions about 1 knob long term vs 2 long term, just want to make sure I know what the right way to address the former is.

Reply:
Alright, if we wanted two knobs permanently, I'd add `include_response_attempt_count`. I had being reading this field as "include the number of request attempts made to the upstream", but now I see that it could be read as "include the number of attempts in a series of retries to an upstream on upstream *requests*". Thus the new field is "include the number of attempts in a series of retries to an upstream on *downstream response*".

Curious about the same question that @alyssawilk has, because if we wanted one knob going forward I would create `include_attempt_count` which would control both including on upstream requests and downstream response, and put the current field on deprecation.

It seems that the only opinion we have in the 1 vs 2 knobs long term is @htuch's vote for two knobs. So I'll go ahead and update this PR to reflect that.

Thank you both :)

Reply:
Conceptually they feel like different enough things to warrant different options I think. The upstream header to me is to let the upstream know specifically which attempt its handling, while the downstream is more informational about how many attempts were made. So +1 from me to make this a second knob.

Reply:
Made it into two knobs. Although both of these knobs are ripe for being folded into [`DebugCofig`](https://github.com/envoyproxy/envoy/blob/master/source/common/router/debug_config.h) once an API is put in place, perhaps for v4 :) cc @mergeconflict 

Also note that I changed names a bit for clarity. The only place I didn't do that was, obviously, in the already existing proto field. I tagged this idea with `// [#next-major-version:]`

Reply:
@htuch I wanted to tag the rename of `include_request_attempt_count` with `// [#next-major-version:]` *only* in the v3 proto but `./tools/proto_format/proto_format.sh fix` keeps dropping it unless I also add it in the v2 proto. Is that intentional?

Reply:
Yeah @junr03 . This is broken IMHO, we shouldn't allow v2 changes to be creating v3 breaking changes. Similar to https://github.com/envoyproxy/envoy/issues/10356. I'll try get some solution here ASAP.

Reply:
Actually, I thought you were suggesting to use `(udpa.annotations.field_migrate).rename` annotations, which are broken as I describe. If all you want is `#next-major-version`, I think it's fine to have it in v2/v3 (for now), while we setup v4alpha and switch to v3 as canonical. So, I think everything in your PR is fine :)

Reply:
Oh, I wasn't aware of `(udpa.annotations.field_migrate).rename`. I like that better, as it is exactly what I would like to happen to that field. But I can leave as is for now, and update in a subsequent PR once the annotation is fixed.

Reply:
I am guessing we prefer using the rename annotation rather than `#next-major-version` as it is more specific.

Reply:
The problem is if we use it on v2 today, it would break v3. We can't make changes only to v3 yet. For this specific PR, I think a `#next-major-version: in the v4 API, we should rename to ...` makes sense. This can live in v2/v3 protos. This is high on the TOFIX list.